docs/DeveloperGuide.md

```diff
@@ -221,7 +222,65 @@ for subsequent prompts.
 ---
 
 ### Parsing User Input and Getting the Right Command
-**_TODO_**: Explain how the app parses user input and determines which `Command` subclass object to instantiate.
+
+![Obtain and Parse User Input](uml/sequenceDiagrams/miscellaneous/images/obtainAndParseUserInput.png)
+
+**(Steps 1 and 2)** When a user enters something into the terminal (when prompted), `UI#getUserInput()` will take in the user's input
+as a `String`, use `String#trim()` to remove leading and trailing whitespaces in the input.
+Thereafter, a line is printed on the terminal to indicate that the user's input has been received
+and will be processed, before returning the user input as a `String` to the calling method (i.e. 
+`WerkIt#startContinuousUserPrompt()`).
+
+**(Step 3)** In `WerkIt#startContinuousUserPrompt()`, the method will pass the obtained user string as a parameter into
+`Parser#parseUserInput()`. The latter method will first check if the user input contains any characters
+or symbols that are deemed as illegal (see [Illegal Characters & Phrases](#illegal-characterse--phrases) for details).
+If at least one illegal character or phrase is found, an `InvalidCommandException` will be thrown and the parsing is
+aborted.
+
+**(Steps 4 to 10)** If no illegal characters and phrases are found, `Parser#parseUserInput()` will examine the first
+word in the user input. This first word should represent the command type that the user wish to execute (i.e. `exercise`,
+`workout`, `plan`, `schedule`, `search`, `help`, or `exit`). Depending on the first word of the user input, different
+methods will be invoked to create the appropriate object of the subclass of the `Command` abstract superclass (see
+example).
```

Comment:
It is unclear what _(see example)_ is referring to as of now or is this to be updated?

Reply:
The intention of the see example was to direct the user to the bullet point after this paragraph:

![image](https://user-images.githubusercontent.com/30099983/161385628-5e52df81-dd7e-4f1a-8026-c24188e04054.png)

If it is confusing, should I rephrase it, or should I remove it completely?

Reply:
Discussed via Telegram. We agreed to rephrase the prompt in parentheses. This change is reflected in Commit 943f96b.